### ISSUE TITLE
Request: spacelift_contexts data source label filter

### ISSUE THREAD

**Issue by dylan-turnbull** [spacelift-io/terraform-provider-spacelift#389]
The `spacelift_contexts` could be improved with the option to supply labels that would be used to filter the returned list of contexts.

While the list can be filtered after being returned by the provider, provider side filtering would result in much cleaner Terraform configuration

```
# An example of what the implementation could look like
data "spacelift_contexts" "contexts" {
  labels = [
    "production",
    "vault"
  ]
}
```

```
# An example of similar functionality in the AWS Terraform provider.
data "aws_subnets" "private" {
  tags = {
    Tier = "Private"
  }
}
```

**Comment by marcinwyszynski** [comment]
Make sense, I'll take this one.

**Comment by marcinwyszynski** [comment]
This is released in 0.1.37.
